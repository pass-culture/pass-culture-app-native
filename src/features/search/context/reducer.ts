import { Hit } from '@algolia/client-search'

import { SearchGroupNameEnumv2 } from 'api/gen'
import { DATE_FILTER_OPTIONS } from 'features/search/enums'
import { addOrRemove, MAX_RADIUS, sortCategories } from 'features/search/helpers/reducer.helpers'
import { SearchState, SearchView } from 'features/search/types'
import { Venue } from 'features/venue/types'
import { AlgoliaVenue } from 'libs/algolia'
import { LocationMode } from 'libs/location/types'
import { SuggestedPlace } from 'libs/place'

export const initialSearchState: SearchState = {
  beginningDatetime: undefined,
  date: null,
  endingDatetime: undefined,
  hitsPerPage: 20,
  locationFilter: { locationType: LocationMode.EVERYWHERE },
  offerCategories: [],
  offerSubcategories: [],
  offerIsDuo: false,
  offerIsFree: false,
  offerIsNew: false,
  isDigital: false,
  priceRange: null,
  query: '',
  tags: [],
  timeRange: null,
  view: SearchView.Landing,
  venue: undefined,
}

export type Action =
  | { type: 'INIT' }
  | { type: 'SET_STATE_FROM_DEFAULT'; payload: Partial<SearchState> }
  | { type: 'SET_STATE'; payload: Partial<SearchState> }
  | { type: 'PRICE_RANGE'; payload: SearchState['priceRange'] }
  | { type: 'SET_MIN_PRICE'; payload: string }
  | { type: 'SET_MAX_PRICE'; payload: string }
  | { type: 'RADIUS'; payload: number }
  | { type: 'TIME_RANGE'; payload: SearchState['timeRange'] }
  | { type: 'TOGGLE_IS_DIGITAL'; payload: SearchState['isDigital'] }
  | { type: 'SET_CATEGORY'; payload: SearchGroupNameEnumv2[] }
  | { type: 'TOGGLE_CATEGORY'; payload: SearchGroupNameEnumv2 }
  | { type: 'TOGGLE_OFFER_FREE' }
  | { type: 'TOGGLE_OFFER_DUO' }
  | { type: 'TOGGLE_OFFER_NEW' }
  | { type: 'TOGGLE_DATE' }
  | { type: 'TOGGLE_HOUR' }
  | { type: 'SELECT_DATE_FILTER_OPTION'; payload: DATE_FILTER_OPTIONS }
  | { type: 'SELECT_DATE'; payload: Date }
  | { type: 'SET_LOCATION_EVERYWHERE' }
  | { type: 'SET_LOCATION_AROUND_ME'; payload?: number }
  | { type: 'SET_LOCATION_PLACE'; payload: { aroundRadius?: number; place: SuggestedPlace } }
  | { type: 'SET_VENUE'; payload?: Venue }
  | { type: 'SET_QUERY'; payload: string }
  | { type: 'SET_VENUES'; payload: Hit<AlgoliaVenue>[] }
  | {
      type: 'SET_LOCATION_FILTERS'
      payload: { locationFilter: SearchState['locationFilter'] }
    }

export const searchReducer = (state: SearchState, action: Action): SearchState => {
  switch (action.type) {
    case 'INIT':
      return { ...initialSearchState, view: state.view }
    case 'SET_STATE_FROM_DEFAULT':
      return { ...initialSearchState, ...action.payload }
    case 'SET_STATE':
      return {
        ...state,
        ...action.payload,
      }
    case 'PRICE_RANGE':
      return { ...state, priceRange: action.payload }
    case 'SET_MIN_PRICE':
      return { ...state, minPrice: action.payload }
    case 'SET_MAX_PRICE':
      return { ...state, maxPrice: action.payload }
    case 'RADIUS':
      if ('aroundRadius' in state.locationFilter) {
        return {
          ...state,
          locationFilter: { ...state.locationFilter, aroundRadius: action.payload },
        }
      } else {
        return state
      }
    case 'TIME_RANGE':
      return { ...state, timeRange: action.payload }
    case 'TOGGLE_CATEGORY':
      return {
        ...state,
        offerCategories: addOrRemove(state.offerCategories, action.payload).sort(sortCategories),
      }
    case 'SET_CATEGORY':
      return { ...state, offerCategories: [...action.payload].sort(sortCategories) }
    case 'TOGGLE_IS_DIGITAL':
      return {
        ...state,
        isDigital: action.payload,
      }
    case 'TOGGLE_OFFER_FREE':
      return { ...state, offerIsFree: !state.offerIsFree }
    case 'TOGGLE_OFFER_DUO':
      return { ...state, offerIsDuo: !state.offerIsDuo }
    case 'TOGGLE_OFFER_NEW':
      return { ...state, offerIsNew: !state.offerIsNew }
    case 'TOGGLE_DATE':
      if (state.date) return { ...state, date: null }
      return {
        ...state,
        date: {
          option: DATE_FILTER_OPTIONS.TODAY,
          selectedDate: new Date().toISOString(),
        },
      }
    case 'TOGGLE_HOUR':
      if (state.timeRange) return { ...state, timeRange: null }
      return {
        ...state,
        timeRange: [8, 24],
      }
    case 'SELECT_DATE_FILTER_OPTION':
      if (!state.date) return state
      return { ...state, date: { ...state.date, option: action.payload } }
    case 'SELECT_DATE':
      if (!state.date) return state
      return { ...state, date: { ...state.date, selectedDate: action.payload.toISOString() } }
    case 'SET_LOCATION_AROUND_ME':
      return {
        ...state,
        locationFilter: {
          locationType: LocationMode.AROUND_ME,
          aroundRadius: action.payload ?? MAX_RADIUS,
        },
      }
    case 'SET_LOCATION_EVERYWHERE':
      return { ...state, locationFilter: { locationType: LocationMode.EVERYWHERE } }
    case 'SET_LOCATION_PLACE':
      return {
        ...state,
        locationFilter: {
          locationType: LocationMode.AROUND_PLACE,
          place: action.payload.place,
          aroundRadius: action.payload.aroundRadius ?? MAX_RADIUS,
        },
      }
    case 'SET_VENUE':
      return {
        ...state,
        venue: action.payload,
      }
    case 'SET_QUERY':
      return { ...state, query: action.payload }
    case 'SET_LOCATION_FILTERS':
      return {
        ...state,
        locationFilter: action.payload.locationFilter,
      }
    default:
      return state
  }
}
