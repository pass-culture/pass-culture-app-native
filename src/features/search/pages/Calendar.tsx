import React from 'react'
import { Text, View } from 'react-native'
import { CalendarList, LocaleConfig } from 'react-native-calendars'

import { Spacer } from 'ui/theme'

LocaleConfig.locales['fr'] = {
  monthNames: [
    'Janvier',
    'Février',
    'Mars',
    'Avril',
    'Mai',
    'Juin',
    'Juillet',
    'Août',
    'Septembre',
    'Octobre',
    'Novembre',
    'Décembre',
  ],
  monthNamesShort: [
    'Janv.',
    'Févr.',
    'Mars',
    'Avril',
    'Mai',
    'Juin',
    'Juil.',
    'Août',
    'Sept.',
    'Oct.',
    'Nov.',
    'Déc.',
  ],
  dayNames: ['Dimanche', 'Lundi', 'Mardi', 'Mercredi', 'Jeudi', 'Vendredi', 'Samedi'],
  dayNamesShort: ['Dim.', 'Lun.', 'Mar.', 'Mer.', 'Jeu.', 'Ven.', 'Sam.'],
}
LocaleConfig.defaultLocale = 'fr'
export const Calendar: React.FC = () => (
  <React.Fragment>
    <Spacer.TopScreen />
    <CalendarList
      minDate={'2021-02-02'}
      maxDate={'2021-12-30'}
      onDayPress={(day) => {
        console.log('selected day', day)
      }}
      monthFormat={'MMMM yyyy'}
      hideDayNames={true}
      firstDay={1}
      disableAllTouchEventsForDisabledDays={true}
      // renderHeader={(date) => <Text>Mois</Text>}
      pastScrollRange={0}
      futureScrollRange={50}
      scrollEnabled={true}
      showScrollIndicator={true}
      markedDates={{
        '2021-02-20': { disabled: true },
      }}
      renderHeader={(date) => <Text>{`${date.getMonth()} ${date.getFullYear()}`}</Text>}
      dayComponent={({ date, state, marking }) => {
        console.log(marking)
        return (
          // <View>
          //   <Text
          //     style={{
          //       textAlign: 'center',
          //       textDecorationLine: 'line-through',
          //       color: state === 'disabled' || marking['disabled'] === true ? 'gray' : 'black',
          //     }}>
          //     {date.day}
          //   </Text>
          //   <Text
          //     style={{
          //       textAlign: 'center',
          //       color: state === 'disabled' || marking['disabled'] === true ? 'gray' : 'black',
          //       fontSize: 10,
          //     }}>
          //     2€
          //   </Text>
          // </View>
          <React.Fragment>
            {state === 'disabled' || marking['disabled'] === true ? (
              <View
                style={{
                  position: 'absolute',
                  transform: [{ rotate: '135deg' }],
                  top: 8,
                  width: 28,
                  height: 1,
                  borderBottomColor: 'gray',
                  borderBottomWidth: 1,
                }}
              />
            ) : null}
            <Text>{date.day}</Text>
            <Text>19.90€</Text>
          </React.Fragment>
        )
      }}
    />
  </React.Fragment>
)
