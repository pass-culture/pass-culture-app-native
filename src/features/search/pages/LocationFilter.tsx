import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import debounce from 'lodash.debounce'
import React, { useRef } from 'react'

import { UseNavigationType } from 'features/navigation/RootNavigator'
import { LocationType } from 'libs/algolia'
import { useGeolocation } from 'libs/geolocation'
import { _ } from 'libs/i18n'
import { Banner, BannerType } from 'ui/components/Banner'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { Spacer } from 'ui/theme'

import { LocationChoice } from '../components/LocationChoice'

import { useSearch } from './SearchWrapper'

const DEBOUNCED_CALLBACK = 500

export const LocationFilter: React.FC = () => {
  const { navigate, goBack } = useNavigation<UseNavigationType>()
  const { position, requestGeolocPermission } = useGeolocation()
  const { dispatch } = useSearch()
  const debouncedGoBack = useRef(debounce(goBack, DEBOUNCED_CALLBACK)).current

  const onPressPickPlace = () => {
    if (debouncedGoBack) debouncedGoBack.cancel()
    navigate('LocationPicker')
  }

  const onPressAroundMe = () => {
    if (position === null) {
      requestGeolocPermission()
    } else {
      dispatch({
        type: 'LOCATION_AROUND_ME',
        payload: { latitude: position.latitude, longitude: position.longitude },
      })
    }
    debouncedGoBack()
  }

  const onPressEverywhere = () => {
    dispatch({ type: 'LOCATION_EVERYWHERE' })
    debouncedGoBack()
  }

  return (
    <React.Fragment>
      <Spacer.TopScreen />
      <Spacer.Column numberOfSpaces={14} />
      <Spacer.Column numberOfSpaces={6} />
      <Banner
        title={_(
          t`Seules les offres Sorties et Physiques seront affichÃ©es pour une recherche avec une localisation`
        )}
        type={BannerType.INFO}
      />
      <Spacer.Column numberOfSpaces={6} />
      <LocationChoice
        testID="pickPlace"
        locationType={LocationType.PLACE}
        arrowNext={true}
        onPress={onPressPickPlace}
      />
      <Spacer.Column numberOfSpaces={4} />
      <LocationChoice
        testID="aroundMe"
        locationType={LocationType.AROUND_ME}
        onPress={onPressAroundMe}
      />
      <Spacer.Column numberOfSpaces={4} />
      <LocationChoice
        testID="everywhere"
        locationType={LocationType.EVERYWHERE}
        onPress={onPressEverywhere}
      />
      <PageHeader title={_(t`Localisation`)} />
    </React.Fragment>
  )
}
