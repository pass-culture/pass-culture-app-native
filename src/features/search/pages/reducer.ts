import { FetchAlgoliaParameters, LocationType, AlgoliaGeolocation } from 'libs/algolia'
import { DATE_FILTER_OPTIONS } from 'libs/algolia/enums'
import { SuggestedPlace } from 'libs/place'

import { clampPrice, addOrRemove } from './reducer.helpers'

export type SearchParameters = Omit<
  FetchAlgoliaParameters,
  'hitsPerPage' | 'page' | 'sortBy' | 'keywords'
>
export type SearchState = SearchParameters & { showResults: boolean }

export const initialSearchState: SearchState = {
  aroundRadius: null,
  offerCategories: [],
  tags: [],
  offerIsDuo: false,
  offerIsFree: false,
  offerIsNew: false,
  offerTypes: {
    isDigital: false,
    isEvent: false,
    isThing: false,
  },
  beginningDatetime: null,
  endingDatetime: null,
  priceRange: null,
  locationType: LocationType.EVERYWHERE,
  geolocation: null,
  date: null,
  timeRange: null,
  showResults: false,
  place: null,
}

export const DEFAULT_TIME_RANGE = [8, 24]

export type Action =
  | { type: 'INIT' }
  | { type: 'INIT_FROM_SEE_MORE'; payload: Partial<SearchState> }
  | { type: 'PRICE_RANGE'; payload: SearchState['priceRange'] }
  | { type: 'TIME_RANGE'; payload: SearchState['timeRange'] }
  | { type: 'OFFER_TYPE'; payload: keyof SearchState['offerTypes'] }
  | { type: 'SHOW_RESULTS'; payload: boolean }
  | { type: 'SET_CATEGORY'; payload: SearchState['offerCategories'] }
  | { type: 'TOGGLE_CATEGORY'; payload: string }
  | { type: 'TOGGLE_OFFER_FREE' }
  | { type: 'TOGGLE_OFFER_DUO' }
  | { type: 'TOGGLE_OFFER_NEW' }
  | { type: 'TOGGLE_DATE' }
  | { type: 'TOGGLE_HOUR' }
  | { type: 'SELECT_DATE_FILTER_OPTION'; payload: DATE_FILTER_OPTIONS }
  | { type: 'SELECT_DATE'; payload: Date }
  | { type: 'LOCATION_AROUND_ME'; payload: AlgoliaGeolocation }
  | { type: 'LOCATION_EVERYWHERE' }
  | { type: 'LOCATION_PLACE'; payload: SuggestedPlace }

export const searchReducer = (state: SearchState, action: Action): SearchState => {
  switch (action.type) {
    case 'INIT':
      return initialSearchState
    case 'SHOW_RESULTS':
      return { ...state, showResults: action.payload }
    case 'INIT_FROM_SEE_MORE':
      return {
        ...state,
        ...action.payload,
        priceRange: clampPrice(action.payload.priceRange),
      }
    case 'PRICE_RANGE':
      return { ...state, priceRange: action.payload }
    case 'TIME_RANGE':
      return { ...state, timeRange: action.payload }
    case 'TOGGLE_CATEGORY':
      return {
        ...state,
        offerCategories: addOrRemove(state.offerCategories, action.payload),
      }
    case 'SET_CATEGORY':
      return { ...state, offerCategories: action.payload }
    case 'OFFER_TYPE':
      return {
        ...state,
        offerTypes: { ...state.offerTypes, [action.payload]: !state.offerTypes[action.payload] },
      }
    case 'TOGGLE_OFFER_FREE':
      return { ...state, offerIsFree: !state.offerIsFree }
    case 'TOGGLE_OFFER_DUO':
      return { ...state, offerIsDuo: !state.offerIsDuo }
    case 'TOGGLE_OFFER_NEW':
      return { ...state, offerIsNew: !state.offerIsNew }
    case 'TOGGLE_DATE':
      if (state.date) return { ...state, date: null }
      return {
        ...state,
        date: {
          option: DATE_FILTER_OPTIONS.TODAY,
          selectedDate: new Date(),
        },
      }
    case 'TOGGLE_HOUR':
      if (state.timeRange) return { ...state, timeRange: null }
      return {
        ...state,
        timeRange: [8, 24],
      }
    case 'SELECT_DATE_FILTER_OPTION':
      if (!state.date) return state
      return { ...state, date: { ...state.date, option: action.payload } }
    case 'SELECT_DATE':
      if (!state.date) return state
      return { ...state, date: { ...state.date, selectedDate: action.payload } }
    case 'LOCATION_AROUND_ME':
      return {
        ...state,
        locationType: LocationType.AROUND_ME,
        geolocation: action.payload,
        place: null,
      }
    case 'LOCATION_EVERYWHERE':
      return {
        ...state,
        locationType: LocationType.EVERYWHERE,
        geolocation: null,
        place: null,
      }
    case 'LOCATION_PLACE':
      return {
        ...state,
        locationType: LocationType.PLACE,
        geolocation: action.payload.geolocation,
        place: action.payload,
      }
    default:
      return state
  }
}
