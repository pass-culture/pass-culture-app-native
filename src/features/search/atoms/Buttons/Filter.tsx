import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import React from 'react'
import LinearGradient from 'react-native-linear-gradient'
import styled from 'styled-components/native'

import { UseNavigationType } from 'features/navigation/RootNavigator'
import { useSearch } from 'features/search/pages/SearchWrapper'
import { getFilterCount } from 'features/search/utils/getFilterCount'
import { _ } from 'libs/i18n'
import { Filter as FilterIcon } from 'ui/svg/icons/Filter'
import { ColorsEnum, UniqueColors, getSpacing, Spacer, Typo } from 'ui/theme'
import { ACTIVE_OPACITY } from 'ui/theme/colors'
import { BorderRadiusEnum } from 'ui/theme/grid'

export const Filter: React.FC = () => {
  const { navigate } = useNavigation<UseNavigationType>()
  const { searchState } = useSearch()
  const filterCount = getFilterCount(searchState)

  return (
    <Container onPress={() => navigate('SearchFilter')} testID="FilterButton">
      <StyledLinearGradient colors={[ColorsEnum.PRIMARY, UniqueColors.FILTER_BUTTON]} angle={106}>
        <FilterIcon color={ColorsEnum.WHITE} />
        <Spacer.Row numberOfSpaces={1} />
        <Title>{_(t`Filtrer`)}</Title>
        <Spacer.Row numberOfSpaces={2} />
        {filterCount > 0 && (
          <React.Fragment>
            <Spacer.Row numberOfSpaces={1} />
            <WhiteBackgroundContainer>
              <Spacer.Row numberOfSpaces={0.5} />
              <Typo.ButtonText color={ColorsEnum.PRIMARY}>{filterCount}</Typo.ButtonText>
            </WhiteBackgroundContainer>
            <Spacer.Row numberOfSpaces={1} />
          </React.Fragment>
        )}
      </StyledLinearGradient>
    </Container>
  )
}

const Container = styled.TouchableOpacity.attrs(() => ({
  activeOpacity: ACTIVE_OPACITY,
}))({ overflow: 'hidden' })

const StyledLinearGradient = styled(LinearGradient)({
  borderRadius: BorderRadiusEnum.BUTTON,
  alignItems: 'center',
  flexDirection: 'row',
  overflow: 'hidden',
  paddingHorizontal: getSpacing(3),
  height: getSpacing(10),
})

const Title = styled(Typo.ButtonText)({ color: ColorsEnum.WHITE })
const WhiteBackgroundContainer = styled.View({
  flexDirection: 'row',
  backgroundColor: ColorsEnum.WHITE,
  borderRadius: getSpacing(6),
  aspectRatio: '1',
  width: getSpacing(6),
  alignItems: 'center',
  justifyContent: 'center',
})
