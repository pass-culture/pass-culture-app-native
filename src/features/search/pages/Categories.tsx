import { t } from '@lingui/macro'
import { useNavigation } from '@react-navigation/native'
import debounce from 'lodash.debounce'
import React, { useRef } from 'react'
import styled from 'styled-components/native'

import { useSearch } from 'features/search/pages/SearchWrapper'
import { CATEGORY_CRITERIA } from 'libs/algolia/enums'
import { _ } from 'libs/i18n'
import { PageHeader } from 'ui/components/headers/PageHeader'
import { Validate } from 'ui/svg/icons/Validate'
import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
import { ACTIVE_OPACITY } from 'ui/theme/colors'

const ALL = 'ALL'
export const DEBOUNCED_CALLBACK = 500

export const useSelectCategory = (callback: () => void) => {
  const { searchState, dispatch } = useSearch()
  const debouncedCallback = useRef(debounce(callback, DEBOUNCED_CALLBACK)).current

  return {
    isCategorySelected: (category: string) => {
      const [selectedCategory] = [...searchState.offerCategories, ALL]
      return selectedCategory === category
    },
    selectCategory: (category: string) => () => {
      const payload = category === ALL ? [] : [category]
      dispatch({ type: 'SET_CATEGORY', payload })
      debouncedCallback()
    },
  }
}

export const Categories: React.FC = () => {
  const { goBack } = useNavigation()
  const { isCategorySelected, selectCategory } = useSelectCategory(goBack)

  return (
    <React.Fragment>
      <React.Fragment>
        <Container>
          <Spacer.TopScreen />
          <Spacer.Column numberOfSpaces={16} />

          {Object.entries(CATEGORY_CRITERIA).map(([category, { label, icon: Icon }]) => {
            const isSelected = isCategorySelected(category)
            const color2 = isSelected ? ColorsEnum.PRIMARY : ColorsEnum.SECONDARY
            const textColor = isSelected ? ColorsEnum.PRIMARY : ColorsEnum.BLACK

            return (
              <LabelContainer key={category} onPress={selectCategory(category)} testID={category}>
                <Spacer.Row numberOfSpaces={4} />
                <Icon size={getSpacing(12)} color={ColorsEnum.PRIMARY} color2={color2} />
                <Spacer.Row numberOfSpaces={2} />
                <Typo.ButtonText color={textColor}>{label}</Typo.ButtonText>
                <Spacer.Flex />
                {isSelected && <Validate color={ColorsEnum.PRIMARY} size={getSpacing(8)} />}
                <Spacer.Row numberOfSpaces={6} />
              </LabelContainer>
            )
          })}

          <Spacer.Column numberOfSpaces={30} />
        </Container>
      </React.Fragment>

      <PageHeader title={_(t`CatÃ©gories`)} />
    </React.Fragment>
  )
}

const Container = styled.ScrollView({ flex: 1 })

const LabelContainer = styled.TouchableOpacity.attrs({
  activeOpacity: ACTIVE_OPACITY,
})({
  flexDirection: 'row',
  alignItems: 'center',
  marginBottom: getSpacing(4),
})
