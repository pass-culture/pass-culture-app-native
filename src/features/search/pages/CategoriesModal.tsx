import React, { FunctionComponent } from 'react'
import { useWindowDimensions } from 'react-native'
import styled from 'styled-components/native'

import { Categories } from 'features/search/pages/Categories'
import { AppModal } from 'ui/components/modals/AppModal'
import { Close } from 'ui/svg/icons/Close'

interface Props {
  visible: boolean
  dismissModal: () => void
}

const MODAL_HEIGHT_RATIO = 3 / 4

export const CategoriesModal: FunctionComponent<Props> = ({ visible, dismissModal }) => {
  const { height: windowHeight } = useWindowDimensions()
  return (
    <AppModal
      animationOutTiming={1}
      visible={visible}
      title="Catégories"
      leftIconAccessibilityLabel={undefined}
      leftIcon={undefined}
      onLeftIconPress={undefined}
      rightIconAccessibilityLabel={'Ne pas filtrer sur les catégories et retourner aux résultats'}
      rightIcon={Close}
      onRightIconPress={dismissModal}>
      <ModalContent windowHeight={windowHeight} testID="categoriesModal">
        <Categories dismissModal={dismissModal} />
      </ModalContent>
    </AppModal>
  )
}

const ModalContent = styled.View<{ windowHeight: number }>(({ windowHeight }) => ({
  width: '100%',
  minHeight: windowHeight * MODAL_HEIGHT_RATIO,
}))
