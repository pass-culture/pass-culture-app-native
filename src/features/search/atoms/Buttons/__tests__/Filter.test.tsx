import { fireEvent, render } from '@testing-library/react-native'
import React from 'react'

import { navigate } from '__mocks__/@react-navigation/native'
import { initialSearchState } from 'features/search/pages/reducer'

import { Filter } from '../Filter'

const mockSearchState = initialSearchState
const mockDispatch = jest.fn()
jest.mock('features/search/pages/SearchWrapper', () => ({
  useSearch: () => ({
    searchState: mockSearchState,
    dispatch: mockDispatch,
  }),
}))
describe('Filter component', () => {
  afterAll(() => jest.resetAllMocks())

  it('should render correctly', () => {
    const { toJSON } = render(<Filter />)
    expect(toJSON()).toMatchSnapshot()
  })

  it('should navigate to Filter page on pressing', () => {
    const { getByTestId } = render(<Filter />)
    fireEvent.press(getByTestId('FilterButton'))
    expect(navigate).toHaveBeenCalledWith('SearchFilter')
  })
})
