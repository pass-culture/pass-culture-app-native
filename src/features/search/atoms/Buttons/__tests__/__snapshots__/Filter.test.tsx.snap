// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "overflow": "hidden",
    }
  }
  testID="FilterButton"
>
  <BVLinearGradient
    colors={
      Array [
        4290717535,
        4284091776,
      ]
    }
    endPoint={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 24,
          "flexDirection": "row",
          "height": 40,
          "overflow": "hidden",
          "paddingHorizontal": 12,
        },
      ]
    }
  >
    <View
      fill="#ffffff"
      height={32}
      width={32}
    >
      <Text>
        undefined-SVG-Mock
      </Text>
    </View>
    <View
      numberOfSpaces={1}
      style={
        Array [
          Object {
            "width": 4,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 15,
            "lineHeight": 20,
          },
        ]
      }
    >
      Filtrer
    </Text>
    <View
      numberOfSpaces={2}
      style={
        Array [
          Object {
            "width": 8,
          },
        ]
      }
    />
  </BVLinearGradient>
</View>
`;
